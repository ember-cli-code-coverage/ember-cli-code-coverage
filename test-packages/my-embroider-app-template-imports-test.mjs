'use strict';

import { execa } from 'execa';
import setupTestDir, { assertCoverageExists, assertFileIsNotEmpty } from './utils.mjs';
import { describe, it } from 'vitest';

const APP_DIR  = 'my-embroider-app-template-imports';

describe('app coverage generation', function () {
    const env = { COVERAGE: 'true' };

    it(`generates coverage with embroider and template imports`, async function () {
      let buildPath = await setupTestDir(APP_DIR, env, {});

      // Embroider tries to reuse .embroider folder if present during builds.
      // We need make sure previous test runs don't affect newer test runs, as the embroider/core
      // version that generated the folder would be different to the one trying to use it, which can lead to unexpected behaviour.
      // await execa('rm',  ['-rf', '.embroider'], { cwd: buildPath, env });
      await execa('rm',  ['-rf', '.embroider'], { cwd: buildPath, env });

      await execa('npx',  ['ember','test', '--test-port=0'], { cwd: buildPath, env });

      await assertCoverageExists(`${buildPath}/coverage`);
    });
});
