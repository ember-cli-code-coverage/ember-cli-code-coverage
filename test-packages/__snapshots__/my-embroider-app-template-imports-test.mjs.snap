// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app coverage generation > generates coverage with embroider and template imports 1`] = `
{
  "app/app.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "lines": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
    "statements": {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
  },
  "app/components/bar.gjs": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 3,
      "pct": 100,
      "skipped": 0,
      "total": 3,
    },
    "statements": {
      "covered": 3,
      "pct": 100,
      "skipped": 0,
      "total": 3,
    },
  },
  "app/router.js": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
    "statements": {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
  },
  "total": {
    "branches": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "branchesTrue": {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": {
      "covered": 1,
      "pct": 50,
      "skipped": 0,
      "total": 2,
    },
    "lines": {
      "covered": 8,
      "pct": 80,
      "skipped": 0,
      "total": 10,
    },
    "statements": {
      "covered": 8,
      "pct": 80,
      "skipped": 0,
      "total": 10,
    },
  },
}
`;
